src/Commands/generate/ApplicationFactory.php

```diff
@@ -43,13 +47,19 @@ public function create(): Application
 
     public function discover(): array
     {
-        $module_generators = [];
-        $serviceManager = \Drush\Drush::service('service.manager');
-        $module_generators = $serviceManager->getGenerators();
+        $module_generator_classes = [];
+        foreach ($this->container->get('module_handler')->getModuleList() as $moduleId => $extension) {
+            $path = DRUPAL_ROOT . '/' . $extension->getPath() . '/src/Drush/';
+            $module_generator_classes = array_merge(
+                $module_generator_classes,
+                $this->serviceManager->discoverModuleGenerators([$path], "\\Drupal\\" . $moduleId . "\\Drush")
+            );
+        }
+        $module_generators = $this->serviceManager->instantiateServices($module_generator_classes, $this->drush_container, $this->container);
```

Comment:
Generator discovery happens right here, and not during DrupalBoot8.

Reply:
Seems like sitewide should also move out of Bootstrap?

Reply:
Sitewide commands are discovered during preflight, but then they are saved until bootstrap so that they may be constructed via their static `create` method. We have to wait until bootstrap to construct them, because the Drupal container isn't available until then. We could use a setter method for the Drupal container, but it seems much better to follow the static `create` pattern consistently, and not force folks to do DI during their container setter method, which would look awkward.

Sitewite commands that do not have a static create factory are instantiated during preflight via `new` or `createEarly()`.